### PR TITLE
Simplify a lot of the mount tuning code

### LINKED ISSUE #2645: Feature request: Update mount description
Currently we don't have a way to update the description of a mount. If we forget to set it, or don't have an accurate description, we're SOL.

### PR DESCRIPTION
**Pull request by jefferai**
Make leases much simpler by not trying to compare against system
default/max -- a bit of a fool's errand since a change to config files
can flip whether it's even valid.

For extra credit, add the ability to tune description; closes #2645

### CODE REVIEW

File: vault/logical_system.go [hunk 1]
```diff
@@ -1544,40 +1552,52 @@ func (b *SystemBackend) handleTuneWriteCommon(
...
+	lock.Lock()
+	defer lock.Unlock()
+
+	// Check again after grabbing the lock
+	mountEntry = b.Core.router.MatchingMountEntry(path)
+	if mountEntry == nil {
+		b.Backend.Logger().Error("sys: tune failed: no mount entry found", "path", path)
+		return handleError(fmt.Errorf("sys: tune of path '%s' failed: no mount entry found", path))
+	}
+	if mountEntry != nil && !mountEntry.Local && repState == consts.ReplicationSecondary {
```
Comment (briankassouf) [inline vault/logical_system.go]: This should use the HasState function:
```
repState.HasState(consts.ReplicationPerformanceSecondary)
```

File: vault/logical_system.go [hunk 2]
```diff
@@ -1586,7 +1606,31 @@ func (b *SystemBackend) handleTuneWriteCommon(
 		}
 	}
 
-	return nil, nil
+	var resp *logical.Response
```
Comment (briankassouf) [inline vault/logical_system.go]: This doesn't seem to be used anywhere, should we just return `nil, nil` below?

File: vault/logical_system_helpers.go [hunk 3]
```diff
@@ -7,61 +7,31 @@ import (
...
-			if newMax == nil && *newDefault > b.Core.maxLeaseTTL {
-				return fmt.Errorf("new backend default lease TTL of %d greater than system max lease TTL of %d",
-					int(newDefault.Seconds()), int(b.Core.maxLeaseTTL.Seconds()))
-			}
-		} else {
-			if newMax == nil && *newDefault > meConfig.MaxLeaseTTL {
-				return fmt.Errorf("new backend default lease TTL of %d greater than backend max lease TTL of %d",
-					int(newDefault.Seconds()), int(meConfig.MaxLeaseTTL.Seconds()))
-			}
+	case newDefault != zero && newMax != zero:
```
Comment (briankassouf) [inline vault/logical_system_helpers.go]: This switch statement might be cleaner if it only had this case and a default case. Unless you like the verbosity of the other statements + comments.

File: vault/logical_system.go (hunk 1 above)
Comment (briankassouf) [reply vault/logical_system.go]: It was changed in my locking PR, so i think it has sense been updated.

**Comment by calvn** [comment]
LGTM aside from the comments that Brian left.

File: vault/logical_system.go (hunk 2 above)
Comment (jefferai) [reply vault/logical_system.go]: Ah, it was used for a warning before but that code is now removed, will fix it.

File: vault/logical_system_helpers.go (hunk 3 above)
Comment (jefferai) [reply vault/logical_system_helpers.go]: In this instance I found it useful to think through the different cases and leave appropriate comments.

**Comment by jefferai** [comment]
Travis failure is from intermittent Cassandra failure; merging.

**Pull request merged**